core/src/main/java/org/apache/carbondata/core/datastore/block/TableBlockInfo.java

```diff
@@ -307,4 +316,45 @@ public void setVersion(ColumnarFormatVersion version) {
   public void setBlockStorageIdMap(Map<String, String> blockStorageIdMap) {
     this.blockStorageIdMap = blockStorageIdMap;
   }
+
+  public byte[] getSerializedData() throws IOException {
```

Comment:
If we implement serializable we cannot control what to write and what not to write, here we have complete control of how to write the data.  I will create the new interface CarbonWritable like hadoop's Writable and add these methods to there.